src/tools/sfdk/task.h

```diff
@@ -69,6 +69,8 @@ class Task : public QObject
     using QObject::QObject;
     ~Task() override;
 
+    void terminate();
```

Comment:
IIUC, terminate sends the process "TERM" signal. What happens if the process refuses to die?

Reply:
No, Task is a general purpose task abstraction. At this level it is not related to (unix) process management.

Reply:
Imagine that my comment was in RemoveProcess::beginTerminate(), which I believe is eventually called after calling terminate() here.

Reply:
RemoteProcess does not try to blindly "fix" that situation with SIGKILL - that might do more harm than good.

Reply:
I'm not saying it should. But waiting forever is also not good.

Reply:
Waiting forever enables the user to fix the situation somehow and provides feedback when nothing was left running orphaned there. Possible "solutions" include "pkill -9 sfdk" which has lesser potential to harm.

Reply:
Now it would be nice if sfdk handled the SIGTERM signal properly, so one wouldn't have to resort to such desperate measures. But we are drifting away from the topic; I believe you have thought about this at least a little bit, and we can always improve it in the future .